### PR TITLE
Maya: Add MayaPy application.

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
This adds mayapy to the application to be launched from a task.

## Testing notes:
1. Setup project with MayaPy as application
2. Launch MayaPy
3. In console window execute:
```
import maya.standalone
maya.standalone.initialize()
```

### CODE REVIEW

**Comment by tokejepsen** [comment]
Split from https://github.com/ynput/OpenPype/pull/5644

**Comment by BigRoy** [comment]
Having the duplicated `environment` in settings for these does seem a bit unfortunate because you might rarely want to differentiate that with your Maya envs themselves, so now you'd need to manage both?

I know these are for testing but we've always used this [**"Debug Shell" launcher action**](https://github.com/BigRoy/OpenPype/blob/58ad8b71676dfdc0f90a0893d751a9c4a4a66791/openpype/modules/colorbleed/launcher_actions/debug_shell.py) which creates a command line shell with the `os.environ` similar to how you'd usually launch the application. This then works for any of the hosts, and then due to how it sets things up you can just type `mayapy` to launch e.g. MayaPy in the maya environment. Different use case maybe - but maybe it can help testing too in a way.

**Comment by BigRoy** [comment]
> In console window execute `from maya import cmds`.

If I'm not mistaken that actually doesn't do much - the `cmds` module will be empty as far as I know until you run:
```python
import maya.standalone
maya.standalone.initialize()
```
That'd also actually start the OpenPype Maya host install process so might be good to test with too to see if any errors occur with `mayapy`.

File: openpype/settings/defaults/system_settings/applications.json [hunk 1]
```diff
@@ -114,6 +114,65 @@
...
+            },
+            "2023": {
+                "use_python_2": false,
+                "executables": {
+                    "windows": [
+                        "C:\\Program Files\\Autodesk\\Maya2023\\bin\\mayapy.exe"
+                    ],
+                    "darwin": [],
+                    "linux": [
+                        "/usr/autodesk/maya2024/bin/mayapy"
```
Comment (moonyuet) [inline openpype/settings/defaults/system_settings/applications.json:160]:
```suggestion
                        "/usr/autodesk/maya2023/bin/mayapy"
```

**Review by moonyuet — COMMENTED** [review]
Code looks good for me except for the small mistake on the setting for 2023
Is it also gonna be added into the AYON setting?

**Comment by moonyuet** [comment]
> > In console window execute `from maya import cmds`.
> 
> If I'm not mistaken that actually doesn't do much - the `cmds` module will be empty as far as I know until you run:
> 
> ```python
> import maya.standalone
> maya.standalone.initialize()
> ```
> 
> That'd also actually start the OpenPype Maya host install process so might be good to test with too to see if any errors occur with `mayapy`.

I think it might work without `maya.standalone.initialize()`  but I agree that we need to check if any errors occurred when mayapy started.
The screenscapture below is the mayapy executed from the OP launcher.
image

**Comment by BigRoy** [comment]
Without `initialize` the `maya.cmds` module exists, but is empty, e.g. the file command `maya.cmds.file` I believe does not exist until initialized. I might be wrong though. 🗡️

**Comment by moonyuet** [comment]
> Without `initialize` the `maya.cmds` module exists, but is empty, e.g. the file command `maya.cmds.file` I believe does not exist until initialized. I might be wrong though. 🗡️

image
maybe you are right.......

**Review by moonyuet — APPROVED** [review]
image
Looks good for me

File: server_addon/applications/server/applications.json [hunk 2]
```diff
@@ -109,6 +109,55 @@
                 }
             ]
         },
+        "maya": {
```
Comment (antirotor) [inline server_addon/applications/server/applications.json:112]: this is wrong, it should be `mayapy` - now it is breaking AYON default settings

**Pull request merged**